EU4toV2/Source/V2World/V2World.cpp

```diff
@@ -144,41 +146,17 @@ std::set<std::string> V2World::discoverProvinceFilenames()
 
 void V2World::importProvinceClimates()
 {
-	string filename = theConfiguration.getVic2Path() + "/map/climate.txt";
-	if (!Utils::DoesFileExist(filename))
+	mappers::ClimateMapper climateMapper;
+	std::map<std::string, std::vector<int>> climateMap = climateMapper.getClimateMap();
+	for (const auto& climate : climateMap)
 	{
-		LOG(LogLevel::Warning) << "Could not find file " << filename
-		                       << ", will not load climates.";
-		return;
-	}
-
-        auto climateObj = parser_8859_15::doParseFile(filename);
-	std::vector keywords = {"mild_climate", "temperate_climate",
-	                        "harsh_climate", "inhospitable_climate"};
-	for (const auto& key : keywords)
-	{
-		auto objs = climateObj->getValue(key);
-		if (objs.size() < 2)
+		for (const auto& provID : climate.second)
 		{
-			LOG(LogLevel::Warning)
-			    << "Found " << objs.size() << " objects with key "
-			    << key << ", will not assign this climate.";
-			continue;
-		}
-		auto provList = objs[1];
-		for (int i = 0; i < provList->numTokens(); ++i)
-		{
-			auto provNum = provList->tokenAsInt(i);
-			if (provNum == 0)
-			{
-				continue;
-			}
-			auto* prov = getProvince(provNum.value());
-			if (prov == NULL)
-			{
-				continue;
-			}
-			prov->setClimate(key);
+			if (provID == 0) continue;
+			auto* prov = getProvince(provID);
+			if (prov == NULL) continue;

```

Comment:
Should check against nullptr instead of NULL.

Reply:
done